client/src/router/index.js

```diff
@@ -7,21 +7,24 @@ Vue.use(VueRouter);
 const routes = [
     {
         path: '/',
-        name: 'Home',
-        component: Home
-    },
-    {
-        path: '/about',
-        name: 'About',
-        // route level code-splitting
-        // this generates a separate chunk (about.[hash].js) for this route
-        // which is lazy-loaded when the route is visited.
-        component: () =>
-            import(/* webpackChunkName: "about" */ '../views/About.vue')
-    },
-    {
-        path: '/status',
-        component: () => import('../views/Status.vue')
+        name: 'UserDataProvider',
+        component: () => import('@/components/guard/UserDataProvider'),
```

Comment:
for UserDataProvider and later LoginGuard use already imported component, not function import, but directive, it must be loaded with the page